release-content/0.14/release-notes/13417_implement_the_full_set_of_sort_methods_on_QueryIter.md

```diff
@@ -1 +1,114 @@
+We might want sort our queries to f.e. display a list of things, or fold over our query with some order-dependent math operation.
+In 0.13 a sort could look like this:
+
+```rust
+#[derive(Component, Copy, Clone, Deref)]
+pub struct Attack(pub usize)
+
+fn handle_enemies(enemies: Query<(&Health, &Attack, &Defense)>) {
+    let mut enemies: Vec<_> = enemies.iter().collect();
+    enemies.sort_by_key(|(_, atk, ..)| *atk)
+    for enemy in enemies {
+        work_with(enemy)
+    }
+}
+```
+
+This can get especially unwieldy and repetitive when sorting within multiple systems.
+Even if we always want the same sort, different [`Query`] types make it unreasonably difficult to abstract away as a user!
+To solve this, we implemented new sort methods on the [`QueryIter`] type, turning the example into:
+
+```rust
+// To be used as a sort key, `Attack` now implements Ord.
+#[derive(Component, Copy, Clone, Deref, PartialEq, Eq, PartialOrd, Ord)]
+pub struct Attack(pub usize)
+
+fn handle_enemies(enemies: Query<(&Health, &Attack, &Defense)>) {
+    for enemy in enemies.iter().sort::<&Attack>() {
+        work_with(enemy)
+    }
+}
+```
+
+To sort our query with the `Attack` component, we specify it as the generic parameter to [`sort`].
+If we wished to sort with more than one [`Component`], we can do so, independent of [`Component`] order in the original [`Query`] type: `enemies.iter().sort::<(&Defense, &Attack)>()`
+
+The generic parameter can be thought of as being a [lens] or "subset" of the original query, on which the underlying sort is actually performed. The result is then internally used to return a new sorted query iterator over the original query items.
+With the default [`sort`], the lens has to be fully [`Ord`], like with [`slice::sort`].
+If this is not enough, we also have the counterparts to the remaining 6 sort methods from [`slice`]!
+
+The generic lens argument works the same way as with [`Query::transmute_lens`]. We do not use filters, they are inherited from the original query.
+The [`transmute_lens`] infrastructure has some nice additional features, which allows for this:
+
+```rust
+fn handle_enemies(enemies: Query<(&Health, &Attack, &Defense, &Rarity)>) {
+    for enemy in enemies.iter().sort_unstable::<Entity>() {
+        work_with(enemy)
+    }
+}
+```
+
+Because we can add [`Entity`] to any lens, we can sort by it without including it in the original query!
+
+These sort methods are fully generic over mutability, so can be used on both [`Query::iter`] and [`Query::iter_mut`]! The rest of the of the iterator methods on [`Query`] do not support sorting.
+The sorts return [`QuerySortedIter`], itself an iterator, enabling the use of further iterator adapters on it.
+Further, [`QuerySortedIter`] implements [`DoubleEndedIterator`] while the initial [`QueryIter`] does not. As a consequence, an empty sort can be used to get a more powerful iterator type:
```

Comment:
IMO all of the text between "Further..." and "Keep in mind" should be cut. These are really neat uses, but this is too much detail for the release notes. If it's not in there already, these should be in the docs though!

Reply:
Fair!

Reply:
The section has been removed.